content/blog/2025-05-08-go-client-GA.md

```diff
@@ -0,0 +1,503 @@
+    }
+    defer client.Close()
+
+    tx := api.NewTransaction(client)
+    cmd := tx.GlideClient
+
+    // Queue multiple commands
+    cmd.Set("user:profile", "John Doe")
+    cmd.Set("user:email", "john@example.com")
+    cmd.SAdd("users", "john")
+
+    // Execute transaction
+    results, err := tx.Exec()
+    if err != nil {
+        log.Printf("Transaction failed: %v", err)
+        return
+    }
+
+    // Process results
+    for _, result := range results {
+        fmt.Println(result)
+    }
+```
+
+## Cluster mode Transaction
+
+To create a cluster transaction, initialize the cluster transaction object using `api.NewClusterTransaction(clusterClient)` to create a new cluster transaction with an established cluster client as an argument.
+`cmd := tx.GlideClusterClient` grants access to a transaction-scoped cluster commands interface, allowing you to construct a sequence of operations to be executed.
+
+```go
+    // Cluster Client Transaction
+    clusterConfig := api.NewGlideClusterClientConfiguration().
+        WithAddress(&api.NodeAddress{Host: "cluster1", Port: 7007}).
+        WithAddress(&api.NodeAddress{Host: "cluster2", Port: 7008}).
+        WithAddress(&api.NodeAddress{Host: "cluster3", Port: 7009})
+
+    clusterClient, err := api.NewGlideClusterClient(clusterConfig)
+    if err != nil {
+        log.Fatal("error connecting to cluster: ", err)
+    }
+    defer clusterClient.Close()
+
+    clusterTx := api.NewClusterTransaction(clusterClient)
+    cmd := clusterTx.GlideClusterClient
+
+    // Define routing option
+    var simpleRoute config.Route = config.RandomRoute
+
+    // Create ping options with specific routing
+    pingOpts := options.ClusterPingOptions{
+        PingOptions: &options.PingOptions{
+            Message: "Hello Valkey Cluster",
+        },
+        RouteOption: &options.RouteOption{
+            Route: simpleRoute,
+        },
+    }
+
+    // Queue commands with routing
+    cmd.PingWithOptions(pingOpts)
+    // Execute transaction
+    results, err := clusterTx.Exec()
+    if err != nil {
+        log.Printf("Cluster transaction failed: %v", err)
+        return
+    }
+
+    // Process results
+    for _, result := range results {
+        fmt.Println(result)
+    }
+```
+
+In cluster transactions, the routing option does not enable multi-node transaction execution.
+
+## Transaction design
+
+The transaction implementation in Valkey GLIDE was designed with several key goals, each achieved through specific implementation approaches:
+
+### Idiomatic Go API
+
+The design provides a natural, Go-like interface for transactions through:
+
+- Clean command calling syntax (`cmd.Set(...)`, `cmd.Get(...)`)
+- Use of Go's embedding to inherit behavior from `baseClient`
+- Explicit error handling that follows Go conventions (`result, err := tx.Exec()`)
+
+### Command Transparency
+
+A key design followed is that commands can be used the same way in both regular and transaction contexts:
```

Comment:
'A key design followed' is a bit awkward, feels like it is missing something (Also 'key' means something in the Valkey context, adding to confusion). 

Do you mean a 'design concept', 'design pattern', or 'design principle'?

I like what you called it in the heading. Maybe revise to something like: "A important principle in the design of the client was transparently accessing the command, in both regular and transactions contexts"